bin/dfx-software-sns-aggregator-ci-wasm-url.test

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+for flavour in prod dev; do
+  (
+    echo "The $flavour sns_aggregator URL should be valid."
+    URL="$(dfx-software-sns-aggregator-ci-wasm-url --flavor "$flavour")"
+    [[ "${URL:-}" =~ ^https: ]] || {
+      echo "ERROR: The $flavour sns_aggregator download should be a valid URL: '$URL'"
+      exit 1
+    } >&2
+    case "$flavour" in
+    dev)
+      [[ "${URL}" =~ sns_aggregator_dev.wasm$ ]] || {
+        echo "ERROR: The dev URL should end with sns_aggregator_dev.wasm: $URL"
+        exit 1
+      } >&2
+      ;;
+    prod)
+      [[ "${URL}" =~ sns_aggregator.wasm$ ]] || {
+        echo "ERROR: The dev URL should end with sns_aggregator.wasm: $URL"
+        exit 1
+      } >&2
+      ;;
+    *)
+      {
+        echo "ERROR: Unrecognized flavour: ${flavour:-}"
+        exit 1
+      } >&2
+      ;;
+    esac
+    DOWNLOAD="$(mktemp "sns_aggregator_${flavour}.XXXX")"
+    curl --no-progress-meter --retry 5 --fail -L "$URL" >"$DOWNLOAD"
+    dfx-canister-nm "$DOWNLOAD" | grep -q canister_query || {
+      echo "ERROR: The downloaded file should be a canister wasm file but doesn't appear to be."
+      echo "       It should be:"
+      echo "       - A wasm file"
+      echo "       - With canister query calls"
+      echo "       - Optionally gzipped"
+      echo "Download file type:"
+      file "$DOWNLOAD"
+      echo "File stats:"
+      ls -l "$DOWNLOAD"
```

Comment:
Note: The URL check fails sometimes.  Why, I have no idea.  The URL seems valid, it looks as if curl "succeeds" but the file is invalid.  Of course when one adds diagnostics the problem disappears, but whenever it reappears it would be good to have a lot of details on what is happening.

Reply:
It might be that CI gets a boundary node in the US while you get one in Europe. Maybe try while connecting your VPN to the US?

At least that we had issues like that while you were away where 1 US boundary node was broken and CI failed 33% of the time.